robot/basestation/static/js/elements/navigation/navigation.js

```diff
@@ -1,191 +1,545 @@
+                    }
+                })
             })
-          } else {
-            appendToConsole(
-              'One or more antenna parameters is missing, if you would like antenna directions then please input them'
-            )
-            $('#antenna-inputs').show()
-            $('#antenna-unchanging').hide()
-          }
         })
-      } else {
-        appendToConsole(
-          'One or more antenna parameters is missing, if you would like antenna directions then please input them'
-        )
-        $('#antenna-inputs').show()
-        $('#antenna-unchanging').hide()
-      }
-    })
-
-    goal_latitude.get(function (val) {
-      if (val != null) {
-        $('#goal-latitude').text(val)
-        goal_longitude.get(function (val) {
-          if (val != null) {
-            appendToConsole(
-              'GPS goal parameters already set, displaying directions to the goal'
-            )
-            $('#goal-longitude').text(val)
-            $('#goal-inputs').hide()
-            $('#goal-unchanging').show()
-          } else {
-            appendToConsole(
-              'One or more GPS goal parameters is missing, if you would like directions to the goal then please input them'
-            )
-            $('#goal-inputs').show()
-            $('#goal-unchanging').hide()
-          }
+
+        goal_latitude.get(function(lat) {
+            goal_longitude.get(function(long) {
+                if (lat && long) {
+                    has_gps_goal.set(true)
+                    logInfo('Found goal coordinates to set')
+                    setupGoalStats(lat, long)
+                } else {
+                    logInfo('Goal parameters not found')
+                }
+            })
+        })
+
+        function setupAntennaStats(lat, long, heading) {
+            $('#antenna-stats-lat').text(lat.toFixed(6))
+            $('#antenna-stats-long').text(long.toFixed(6))
+            $('#antenna-stats-heading').text(heading + '°')
+        }
+
+        function setupGoalStats(lat, long) {
+            $('#goal-stats-lat').text(lat.toFixed(6))
+            $('#goal-stats-long').text(long.toFixed(6))
+        }
+    }
+
+    $('#antenna-select-lat-format-btn').on('click', function(event) {
+        event.preventDefault()
+        createAntennaLatitudeHandler()
+    })
+    $('#antenna-select-long-format-btn').on('click', function(event) {
+        event.preventDefault()
+        createAntennaLongitudeHandler()
+    })
+    $('#antenna-confirm-btn').on('click', function(event) {
+        event.preventDefault()
+
+        let lat = -1
+        let long = -1
+
+        try {
+            if ($("#antenna-lat-fieldset").attr('format') == 'DD') {
+                let decdeg = parseFloat(($("#antenna-DD-dec-deg-input.lat").val()))
+                $("#antenna-DD-dec-deg-input.lat").attr("value", decdeg)
+
+                if (isNaN(decdeg)) throw "Bad Input lat DD"
```

Comment:
For consistency sake, please add some curly brackets and put the body on the next line. There are further occurances throughout the code to look out for as well.

Reply:
fixed